### PR TITLE
(CDAP-4924) Moved the table existence check to job setup time from de…

### PR DESCRIPTION
**Pull request by bdmogal**
…ploy time.

Jira: [CDAP-4924](https://issues.cask.co/browse/CDAP-4924)

### CODE REVIEW

File: database-plugins/src/test/java/co/cask/hydrator/plugin/teradata/test/TeradataPluginTestRun.java [hunk 1]
```diff
@@ -358,56 +358,22 @@ public void testNonExistentDBTable() throws Exception {
 
     ETLBatchConfig etlConfig = new ETLBatchConfig("* * * * *", sourceBadName, table, transforms);
     Id.Application appId = Id.Application.from(Id.Namespace.DEFAULT, "dbSourceTest");
```
Comment (albertshau) [inline database-plugins/src/test/java/co/cask/hydrator/plugin/teradata/test/TeradataPluginTestRun.java]: should change the name of this app.  There are travis test failures that seem to indicate some other test uses the same app name.

**Comment by albertshau** [comment]
comment about test failures, otherwise lgtm.  Verified on my cluster that pipelines using these plugins are ok.  Saw permgen get filled, then garbage collected successfully.

File: database-plugins/src/test/java/co/cask/hydrator/plugin/teradata/test/TeradataPluginTestRun.java (hunk 1 above)
Comment (bdmogal) [inline database-plugins/src/test/java/co/cask/hydrator/plugin/teradata/test/TeradataPluginTestRun.java]: will try and make these tests more foolproof. Really need a better way of cleaning up when using TestBase - https://issues.cask.co/browse/CDAP-4925

**Comment by bdmogal** [comment]
@albertshau fixed test failures. Please review again.

**Comment by albertshau** [comment]
:+1:

**Pull request merged**